### PR TITLE
Deberta_v2 tf

### PR DESCRIPTION
**Pull request by kamalkraj**
# What does this PR do?

Deberta-v2 TF

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@LysandreJik @patrickvonplaten

### CODE REVIEW

**Comment by kamalkraj** [comment]
@Rocketknight1 
https://github.com/huggingface/transformers/pull/12972#discussion_r684418611
gather function fails while running `run_glue.py` from examples
Screenshot 2021-08-13 at 9 55 59 PM

If i replace the gather function with experimental NumPy take_along_axis works - https://gist.github.com/kamalkraj/73ad5fa2b84de7e201e05464e11a4fec

**Comment by Rocketknight1** [comment]
Hi @kamalkraj, do you know what shape the inputs are to the gather/take_along_axis? I'm going to try to construct a small test case that fails for my gather function but not for take_along_axis. If you can find a simple test case that fails, feel free to send that too so I can fix the function!

**Comment by kamalkraj** [comment]
Hi @Rocketknight1 
I have tried few tests for `torch.gather ` when you initially shared the function. notebook link- https://colab.research.google.com/drive/1ujI6zKTuuryAO2Nfw9U1ZftyZyC4VUVS?usp=sharing

**Comment by Rocketknight1** [comment]
In all of those cases, it looks like the TF `torch_gather` function gets the same results as the actual `torch.gather`, right? Is there a difference?

**Comment by kamalkraj** [comment]
No. TF `torch_gather` function gets the same output as `torch.gather`.

Actually, in runtime, this branch never gets called https://github.com/huggingface/transformers/blob/e2f07c01e93611fbd96f85204c9a2129bc81862b/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L766

because both query_layer and key_layer are of the same size
https://github.com/huggingface/transformers/blob/e2f07c01e93611fbd96f85204c9a2129bc81862b/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L571-L572

**Comment by kamalkraj** [comment]
Hi @BigBird01,

I was going through `deberta-v2` implementation inside huggingface and as per my understanding, for `deberta-v2` the below branch will be never executed.
https://github.com/huggingface/transformers/blob/e2f07c01e93611fbd96f85204c9a2129bc81862b/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L766

Because query_layer and key_layer shapes are -> 
`[batch_size * num_attention_heads,  sequence_length, attention_head_size] `

the above condition may be needed for `deberta`. Huggingface has separate implementation for `deberta` and `deberta-v2`
if my assumption is correct we can remove those never executed control flow branches from the `deberta-v2` code.

**Comment by BigBird01** [comment]
Yes.  We can remove it to make the code clear.

Thanks!
Pengcheng

**Comment by Rocketknight1** [comment]
> @Rocketknight1
> [#12972 (comment)](https://github.com/huggingface/transformers/pull/12972#discussion_r684418611)
> gather function fails while running `run_glue.py` from examples
> Screenshot 2021-08-13 at 9 55 59 PM
> 
> If i replace the gather function with experimental NumPy take_along_axis works - https://gist.github.com/kamalkraj/73ad5fa2b84de7e201e05464e11a4fec

Hi @kamalkraj, can you share the exact glue task / command you used? I still can't reproduce the bug - I tried this:
```
python run_glue.py --model_name_or_path kamalkraj/deberta-v2-xlarge --task_name mnli --do_train --do_eval --do_predict --output_dir output
```

This seemed to work fine with `torch_gather`.

**Comment by kamalkraj** [comment]
the issue is solved with this commit https://github.com/huggingface/transformers/pull/13120/commits/90c122dedf95e6f4d1ff4395b08783f851e6eb02 .

`torch_gather` function under those `if` condition was creating the issue. I removed those conditions as it was unnecessary .
You can see the discussion https://github.com/huggingface/transformers/pull/13120#issuecomment-899782332

I also opened another pull request to remove from PyTorch model also. https://github.com/huggingface/transformers/pull/13145

**Comment by kamalkraj** [comment]
Hi @Rocketknight1 ,
https://github.com/huggingface/transformers/pull/13145 is merged to master. Now the TF implementation is the same as the torch Implementation. and runs without any issues

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 1]
```diff
@@ -0,0 +1,1667 @@
...
+    input_processing,
+    shape_list,
+)
+from ...utils import logging
+from .configuration_deberta_v2 import DebertaV2Config
+
+
+logger = logging.get_logger(__name__)
+
+# tnp.experimental_enable_numpy_behavior()
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:49]: delete comment maybe?

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 2]
```diff
@@ -0,0 +1,1667 @@
...
+
+_CONFIG_FOR_DOC = "DebertaV2Config"
+_TOKENIZER_FOR_DOC = "DebertaV2Tokenizer"
+_CHECKPOINT_FOR_DOC = "kamalkraj/deberta-v2-xlarge"
+
+TF_DEBERTA_V2_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "kamalkraj/deberta-v2-xlarge",
+    # See all DeBERTa models at https://huggingface.co/models?filter=DeBERTa
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:57]:
```suggestion
    # See all DeBERTa models at https://huggingface.co/models?filter=deberta-v2
```

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 3]
```diff
@@ -0,0 +1,1667 @@
...
+        return outputs
+
+
+class TFDebertaV2ConvLayer(tf.keras.layers.Layer):
+    def __init__(self, config: DebertaV2Config, **kwargs):
+        super().__init__(**kwargs)
+
+        self.kernel_size = getattr(config, "conv_kernel_size", 3)
+        # groups = getattr(config, "conv_groups", 1)
+        self.conv_act = getattr(config, "conv_act", "tanh")
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:276]:
```suggestion
        self.conv_act = get_tf_activation(getattr(config, "conv_act", "tanh"))
```

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 4]
```diff
@@ -0,0 +1,1667 @@
...
+        out = tf.nn.conv2d(
+            tf.expand_dims(hidden_states, 1),
+            tf.expand_dims(self.conv_kernel, 0),
+            strides=1,
+            padding=[[0, 0], [0, 0], [self.padding, self.padding], [0, 0]],
+        )
+        out = tf.squeeze(tf.nn.bias_add(out, self.conv_bias), 1)
+        rmask = tf.cast(1 - input_mask, tf.bool)
+        out = tf.where(tf.broadcast_to(tf.expand_dims(rmask, -1), shape_list(out)), 0.0, out)
+        out = get_tf_activation(self.conv_act)(self.dropout(out, training=training))
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:306]:
```suggestion
        hidden_states = self.dropout(out, training=training)
        hidden_states = self.conv_act(hidden_states)
```

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 5]
```diff
@@ -0,0 +1,1667 @@
...
+        else:
+            input_mask = tf.cast(tf.math.reduce_sum(attention_mask, axis=-2) > 0, dtype=tf.uint8)
+
+        all_hidden_states = () if output_hidden_states else None
+        all_attentions = () if output_attentions else None
+
+        attention_mask = self.get_attention_mask(attention_mask)
+        relative_pos = self.get_rel_pos(hidden_states, query_states, relative_pos)
+
+        if isinstance(hidden_states, Sequence):
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:407]: Can this be true? That `hidden_states` is a `Sequence`?

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 6]
```diff
@@ -0,0 +1,1667 @@
...
+                training=training,
+            )
+            output_states = layer_outputs[0]
+
+            if i == 0 and self.conv is not None:
+                output_states = self.conv(hidden_states, output_states, input_mask)
+
+            if query_states is not None:
+                query_states = output_states
+                if isinstance(hidden_states, Sequence):
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:435]: same here when is `hidden_states` of type `Sequence`?

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 7]
```diff
@@ -0,0 +1,1667 @@
...
+    ]
+    return tf.broadcast_to(c2p_pos, shapes)
+
+
+def pos_dynamic_expand(pos_index, p2c_att, key_layer):
+    shapes = shape_list(p2c_att)[:2] + [shape_list(pos_index)[-2], shape_list(key_layer)[-2]]
+    return tf.broadcast_to(pos_index, shapes)
+
+
+def torch_gather(x, indices, gather_axis):
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:524]: (nit) not a huge fan of the name - can we call it something like `gather_by_index` maybe?

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 8]
```diff
@@ -0,0 +1,1667 @@
...
+        super().__init__(**kwargs)
+        self.predictions = TFDebertaV2LMPredictionHead(config, input_embeddings, name="predictions")
+
+    def call(self, sequence_output: tf.Tensor) -> tf.Tensor:
+        prediction_scores = self.predictions(hidden_states=sequence_output)
+
+        return prediction_scores
+
+
+# @keras_serializable
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:1013]: Can we delete the comment here? :-)

File: tests/test_modeling_tf_deberta_v2.py [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+
+
+@require_tf
+class TFDeBERTaV2ModelIntegrationTest(unittest.TestCase):
+    @unittest.skip(reason="Model not available yet")
+    def test_inference_masked_lm(self):
+        pass
+
+    @slow
+    def test_inference_no_head(self):
```
Comment (patrickvonplaten) [inline tests/test_modeling_tf_deberta_v2.py:269]: very nice!

**Review by patrickvonplaten — APPROVED** [review]
Very nice work @kamalkraj! I only left some nits - looks good to me to merge! 

The only thing that confused me a bit was the if-else logic depending on whether `hidden_states` is of type sequences, *e.g.* here: https://github.com/huggingface/transformers/pull/13120/files#r694681300
-> when would that be the case?

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 1 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: yes. Done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 2 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 3 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 4 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 5 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: hidden_states is never a type of sequence. 
removed

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 6 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: `hidden_states` is never a type of `sequence`. and query_states is always `None`
removed code block

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 7 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: renamed to `take_along_axis`. name from `numpy` function

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 8 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: done

File: tests/test_modeling_tf_deberta_v2.py (hunk 9 above)
Comment (kamalkraj) [reply tests/test_modeling_tf_deberta_v2.py:269]: thanks

**Comment by kamalkraj** [comment]
Hi @patrickvonplaten ,
thanks for the review.
committed changes.

File: docs/source/index.rst [hunk 10]
```diff
@@ -345,7 +345,7 @@ Flax), PyTorch, and/or TensorFlow.
 +-----------------------------+----------------+----------------+-----------------+--------------------+--------------+
 |           DeBERTa           |       ✅       |       ✅       |       ✅        |         ✅         |      ❌      |
 +-----------------------------+----------------+----------------+-----------------+--------------------+--------------+
-|         DeBERTa-v2          |       ✅       |       ❌       |       ✅        |         ❌         |      ❌      |
+|         DeBERTa-v2          |       ✅       |       ❌       |       ✅        |         ✅         |      ❌      |
```
Comment (LysandreJik) [inline docs/source/index.rst:348]: :heart_eyes:

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 11]
```diff
@@ -0,0 +1,1658 @@
...
+
+logger = logging.get_logger(__name__)
+
+
+_CONFIG_FOR_DOC = "DebertaV2Config"
+_TOKENIZER_FOR_DOC = "DebertaV2Tokenizer"
+_CHECKPOINT_FOR_DOC = "kamalkraj/deberta-v2-xlarge"
+
+TF_DEBERTA_V2_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "kamalkraj/deberta-v2-xlarge",
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:55]: We would need to migrate the TF checkpoint to the official one in `microsoft/deberta-v2-xlarge`

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 12]
```diff
@@ -0,0 +1,1658 @@
...
+                    ],
+                ),
+                -1,
+            )
+            score += p2p_att
+
+        return score
+
+
+class TFDebertaV2Embeddings(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:824]: This looks identical to the `TFDebertaEmbeddings` that you have implemented in the TF DeBERTa implementation. Would you mind adding a comment mentioning it's a direct copy? We have CI in place to ensure that the two copies do not diverge
```suggestion
# Copied from transformers.models.deberta.modeling_deberta.TFDebertaEmbeddings with xxx->yyy
class TFDebertaV2Embeddings(tf.keras.layers.Layer):
```
You can replace the `xxx` and `yyy` with the appropriate changes, for example here I believe the `StableDropout` changes so you would put
```suggestion
# Copied from transformers.models.deberta.modeling_deberta.TFDebertaEmbeddings with TFDebertaStableDropout->TFDebertaV2StableDropout
class TFDebertaV2Embeddings(tf.keras.layers.Layer):
```

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 13]
```diff
@@ -0,0 +1,1658 @@
...
+
+    def call(self, hidden_states: tf.Tensor, input_tensor: tf.Tensor, training: bool = False) -> tf.Tensor:
+        hidden_states = self.dense(inputs=hidden_states)
+        hidden_states = self.dropout(hidden_states, training=training)
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+
+        return hidden_states
+
+
+class TFDebertaV2Layer(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:232]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 14]
```diff
@@ -0,0 +1,1658 @@
...
+            self.intermediate_act_fn = config.hidden_act
+
+    def call(self, hidden_states: tf.Tensor) -> tf.Tensor:
+        hidden_states = self.dense(inputs=hidden_states)
+        hidden_states = self.intermediate_act_fn(hidden_states)
+
+        return hidden_states
+
+
+class TFDebertaV2Output(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:214]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 15]
```diff
@@ -0,0 +1,1658 @@
...
+        attention_output = self.dense_output(
+            hidden_states=self_outputs[0], input_tensor=query_states, training=training
+        )
+
+        output = (attention_output,) + self_outputs[1:]
+
+        return output
+
+
+class TFDebertaV2Intermediate(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:194]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 16]
```diff
@@ -0,0 +1,1658 @@
...
+        self.dropout = TFDebertaV2StableDropout(config.hidden_dropout_prob, name="dropout")
+
+    def call(self, hidden_states, input_tensor, training: bool = False):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states, training=training)
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+        return hidden_states
+
+
+class TFDebertaV2Attention(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:157]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 17]
```diff
@@ -0,0 +1,1658 @@
...
+        super().__init__(**kwargs)
+        self.drop_prob = tf.convert_to_tensor(drop_prob, dtype=tf.float32)
+
+    def call(self, inputs: tf.Tensor, training: tf.Tensor = False):
+        if training and self.drop_prob > 0:
+            return TFDebertaV2XDropout(inputs, self.drop_prob)
+        return inputs
+
+
+class TFDebertaV2SelfOutput(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:143]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 18]
```diff
@@ -0,0 +1,1658 @@
...
+
+    def custom_grad(upstream_grad):
+        return tf.cond(
+            scale > 1, lambda: (tf.where(mask, 0.0, upstream_grad) * scale, None), lambda: (upstream_grad, None)
+        )
+
+    return input, custom_grad
+
+
+class TFDebertaV2StableDropout(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:125]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 19]
```diff
@@ -0,0 +1,1658 @@
...
+
+def get_mask(input, dropout):
+    mask = tf.cast(
+        1 - tf.compat.v1.distributions.Bernoulli(probs=1 - dropout).sample(sample_shape=shape_list(input)), tf.bool
+    )
+    return mask, dropout
+
+
+@tf.custom_gradient
+def TFDebertaV2XDropout(input, local_ctx):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:112]: Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 20]
```diff
@@ -0,0 +1,1658 @@
...
+                mask = tf.cast(tf.expand_dims(mask, axis=2), tf.float32)
+
+            final_embeddings = final_embeddings * mask
+
+        final_embeddings = self.dropout(final_embeddings, training=training)
+
+        return final_embeddings
+
+
+class TFDebertaV2PredictionHeadTransform(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:928]: Same remark regarding the `# Copied from` statement (see `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 21]
```diff
@@ -0,0 +1,1658 @@
...
+
+    def call(self, hidden_states: tf.Tensor) -> tf.Tensor:
+        hidden_states = self.dense(inputs=hidden_states)
+        hidden_states = self.transform_act_fn(hidden_states)
+        hidden_states = self.LayerNorm(hidden_states)
+
+        return hidden_states
+
+
+class TFDebertaV2LMPredictionHead(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:952]: Same remark regarding the `# Copied from` statement (see `TFDebertaV2Embeddings` comment for more details)

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 22]
```diff
@@ -0,0 +1,1658 @@
...
+        seq_length = shape_list(hidden_states)[1]
+        hidden_states = tf.reshape(tensor=hidden_states, shape=[-1, self.hidden_size])
+        hidden_states = tf.matmul(a=hidden_states, b=self.input_embeddings.weight, transpose_b=True)
+        hidden_states = tf.reshape(tensor=hidden_states, shape=[-1, seq_length, self.vocab_size])
+        hidden_states = tf.nn.bias_add(value=hidden_states, bias=self.bias)
+
+        return hidden_states
+
+
+class TFDebertaV2OnlyMLMHead(tf.keras.layers.Layer):
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:995]: Same remark regarding the `# Copied from` statement (see `TFDebertaV2Embeddings` comment for more details)

**Review by LysandreJik — COMMENTED** [review]
Tremendous work putting this all together @kamalkraj!

Most of my comments are regarding the `# Copied from` statements that could be added to most classes here, it seems even the actual model classes like `TFDebertaV2ForSequenceClassification` could benefit from them.

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 11 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:55]: Okay

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 12 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:835]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 18 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:130]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 19 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:116]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 21 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:965]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 22 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:1009]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 13 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:242]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 14 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:223]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 15 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:202]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 16 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:164]: done

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 17 above)
Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:149]: done

**Comment by kamalkraj** [comment]
Hi @LysandreJik,
committed changes.

**Review by Rocketknight1 — APPROVED** [review]
Overall, this looks like a very solid PR now, and if tests are passing with good performance then I think it should be just about ready to go, assuming everyone else is in agreement!

**Review by LysandreJik — APPROVED** [review]
Thanks for your work @kamalkraj!

**Comment by sh0416** [comment]
Is this code compatible with model.fit?

**Pull request merged**